### PR TITLE
add burn execution tests

### PR DESCRIPTION
**Pull request by ChihChengLiang**

### CODE REVIEW

File: contracts/BurnExecution.sol [hunk 1]
```diff
@@ -95,51 +93,61 @@ contract BurnExecution is FraudProofHelpers {
...
     }
 
+    function ApplyBurnExecutionTx(
+        Types.AccountMerkleProof memory _fromAccountProof,
+        Types.BurnExecution memory _tx
+    ) public view returns (bytes memory updatedAccount, bytes32 newRoot) {
+        Types.UserAccount memory account = _fromAccountProof.accountIP.account;
+
+        account.balance -= account.burn;
+        account.burn = 0;
```
Comment (ChihChengLiang) [inline contracts/BurnExecution.sol:103]: @barryWhiteHat do we want to reset burn to 0?
It's a "monthly burn" and the spec does not reset to 0. 
If we don't want to reset to zero, what do we do if we don't have enough balance the next month?

File: contracts/BurnExecution.sol [hunk 2]
```diff
@@ -95,51 +93,61 @@ contract BurnExecution is FraudProofHelpers {
...
      * @notice Overrides processTx in FraudProof
      */
-    function processTx(
+    function processBurnExecutionTx(
         bytes32 _balanceRoot,
-        bytes32 _accountsRoot,
         Types.BurnExecution memory _tx,
-        Types.PDAMerkleProof memory _from_pda_proof,
-        Types.AccountProofs memory accountProofs
+        Types.AccountMerkleProof memory _fromAccountProof
```
Comment (ChihChengLiang) [inline contracts/BurnExecution.sol:117]: @vaibhavchellani I removed pda proof and the pubkey tree root for now. We might want to add it back depending on how we do the signature validation.

File: contracts/BurnExecution.sol (hunk 1 above)
Comment (barryWhiteHat) [reply contracts/BurnExecution.sol:103]: So we don't reset to 0. if a user wants to do this they can use authorize burn with amount 0.

If they don't have enough balance the coordinator just does not include them in the batch for that month.

Comment (ChihChengLiang) [reply contracts/BurnExecution.sol:103]: Sounds good. Let me remove the reset and add a balance check.

**Pull request merged**